core-java-modules/core-java-14/src/test/java/com/baeldung/java14/foreign/api/ForeignMemoryUnitTest.java

```diff
@@ -0,0 +1,74 @@
+package com.baeldung.java14.foreign.api;
+
+import jdk.incubator.foreign.*;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.BlockJUnit4ClassRunner;
+
+import java.lang.invoke.VarHandle;
+import java.nio.ByteOrder;
+
+import static org.hamcrest.core.Is.*;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertThat;
+
+public class ForeignMemoryUnitTest {
+
+    @Test
+    public void whenValueIsSetThenAccessTheValue(){
+        long value = 10;
+        MemoryAddress memoryAddress = MemorySegment.allocateNative(8).baseAddress();
+        VarHandle varHandle = MemoryHandles.varHandle(long.class, ByteOrder.nativeOrder());
+        varHandle.set(memoryAddress, value);
+        assertThat(varHandle.get(memoryAddress), is(value));
+    }
+
+    @Test
+    public void whenMultipleValuesAreSetThenAccessAll(){
+        VarHandle varHandle = MemoryHandles.varHandle(int.class, ByteOrder.nativeOrder());
+        try(MemorySegment memorySegment = MemorySegment.allocateNative(100)){
+            MemoryAddress base = memorySegment.baseAddress();
+            for(int i=0; i<25; i++){
+                varHandle.set(base.addOffset((i*4)), i);
+            }
+
+            for(int i=0; i<25; i++){
+                assertThat(varHandle.get(base.addOffset((i*4))), is(i));
+            }
+        }
+    }
+
+    @Test
+    public void testMemoryLayout(){
```

Comment:
Unit tests with names that start `test` often don't explain themselves well. Let's stick with the `given_when_then` style. Maybe `whenSetValuesWithMemoryLayout_thenTheyCanBeRetrieved`

Reply:
Updated all the unit test method names with the `given_when_then`  style.